### DISCUSSION TITLE
Zephyr crashes in heap

### DISCUSSION THREAD

**Discussion by EduardoGoncalves1966** [zephyrproject-rtos/zephyr#77671] [Q&A]
Hi

I have trying to debug a program using Ozone and J-Link Ultra+.
The program is crashing with a memory fault (Ozone indicates a Stack overflow) and the LR register has the value 0xFFFFFFBC.
This happens while allocating memory during a call to zbus_vded_exec.

Does anybody have any ideas how to find out the reason for this?

thanks

**Comment by yashi** [comment]
Increase your thread's stack size. If it's the main thread, do:

```
west build ... -- -DCONFIG_MAIN_STACK_SIZE=20000
```
Or so.

If it's not stackoverflow, use `gdb` to see what's wrong.
https://docs.zephyrproject.org/latest/develop/west/build-flash-debug.html#debugging-west-debug-west-debugserver

**Comment by EduardoGoncalves1966** [comment]
Hi, thanks for the answer.
I have increased stack size and heap size but the problem remains.
Does GDB allow a better understanding of the problem?
I am have been using a complex environment, Visual Studio Code in Windows and zephyr building system running in VMbox with Ubuntu.
GDB was also used but it didn´t provide much information.
So I started using Ozone in Ubunto directly so I got some more information.
thanks

**Reply by yashi** [reply]
If the memory fault is real, you can get backtrace with GDB.
